test/test_flake8.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from pathlib import Path
+import sys
+
+from flake8 import LOG
+from flake8.api.legacy import get_style_guide
+
+
+# Avoid debug and info messages from flake8 internals
+LOG.setLevel(logging.WARN)
+
+
+def test_flake8():
+    style_guide = get_style_guide(
+        ignore=['D100', 'D104'],
+        show_source=True,
+    )
+    style_guide_tests = get_style_guide(
+        ignore=['D100', 'D101', 'D102', 'D103', 'D104', 'D105', 'D107'],
+        show_source=True,
+    )
+
+    stdout = sys.stdout
+    sys.stdout = sys.stderr
+
+    report = style_guide.check_files([
+        str(Path(__file__).parents[1] / 'colcon_sanitizer_reports'),
+    ])
+    report_tests = style_guide_tests.check_files([
+        str(Path(__file__).parents[1] / 'test'),
+    ])
+    sys.stdout = stdout
+
+    total_errors = report.total_errors + report_tests.total_errors
+    if total_errors:
+        # Output summary with per-category counts
+        print()
+        report._application.formatter.show_statistics(report._stats)
```

Comment:
The logic of this line is wrong since `report.total_errors` might be zero. See the updated logic in other packages, e.g. https://github.com/colcon/colcon-core/blob/7dad6d70253836411614d48a9329d7eeaf7accb0/test/test_flake8.py#L47-L51